cmd/zk_quorum_check.go

```diff
@@ -0,0 +1,132 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/dcos/dcos-checks/client"
+	"github.com/dcos/dcos-go/dcos"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// zookeeperQuorumCmd represents the zookeeper-quorum command
+var zookeeperQuorumCmd = &cobra.Command{
+	Use:   "zk-quorum",
+	Short: "DC/OS check zookeeper in quorum",
+	Long:  `Zookeeper quorum`,
+	Run: func(cmd *cobra.Command, args []string) {
+		RunCheck(NewZookeeperQuorumCheck("DC/OS zookeeper quorum check"))
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(zookeeperQuorumCmd)
+}
+
+// ZkResponse struct response for localhost:8181/exhibitor/v1/cluster/state/<host>
+type ZkResponse struct {
+	Response struct {
+		Switches struct {
+			Restarts bool `json:"restarts"`
+			Cleanup  bool `json:"cleanup"`
+			Backups  bool `json:"backups"`
+		} `json:"switches"`
+		State       int    `json:"state"`
+		Description string `json:"description"`
+		IsLeader    bool   `json:"isLeader"`
+	} `json:"response"`
+	ErrorMessage string `json:"errorMessage"`
+	Success      bool   `json:"success"`
+}
+
+// ZookeeperQuorumCheck struct
+type ZookeeperQuorumCheck struct {
+	Name string
+}
+
+// Run the check
+func (zk *ZookeeperQuorumCheck) Run(ctx context.Context, cfg *CLIConfigFlags) (string, int, error) {
+	if cfg.Role != dcos.RoleMaster {
+		return "", statusFailure, fmt.Errorf("Check can be run only on masters")
+	}
+
+	httpClient, err := client.NewClient(cfg.IAMConfig, cfg.CACert)
+	if err != nil {
+		return "", statusUnknown, errors.Wrap(err, "unable to create HTTP client")
+	}
+
+	url, err := zk.getURL(httpClient, cfg)
+	if err != nil {
+		return "", 0, err
+	}
+	logrus.Debugf("GET %s", url)
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return "", statusUnknown, errors.Wrap(err, "unable to create a new HTTP request")
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return "", statusUnknown, errors.Wrapf(err, "unable to execute GET %s", healthURLPrefix)
+	}
+	defer resp.Body.Close()
+
+	var zr ZkResponse
+	if err := json.NewDecoder(resp.Body).Decode(&zr); err != nil {
+		return "", statusUnknown, errors.Wrap(err, "unable to unmarshal exhibitor response")
+	}
+
+	return zr.Response.Description, 1, errors.Wrap(nil, zr.ErrorMessage)
+}
+
+// ID returns a unique check identifier.
+func (zk *ZookeeperQuorumCheck) ID() string {
+	return zk.Name
+}
+
+func (zk *ZookeeperQuorumCheck) getURL(httpClient *http.Client, cfg *CLIConfigFlags) (*url.URL, error) {
+	ip, err := cfg.IP(httpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	host := ip.String()
+	port := 0
```

Comment:
`port` here is only used in the block defined by `if cfg.ForceTLS`.  let's move it into that block.